File: ruby/Composite/structure.rb
require_relative 'group'

# Collection/group of blocks
class Structure
  # Collection of child group
  def initialize
    @groupList = Array.new
  end

  # assemble method should treat both single and group of blocks same.
  def assemble
    for group in @groupList
      group.assemble
    end
    print "Creating a structure from Group objects.\n"
  end

  # adds the group to the structure
  def add(group)
    @groupList.push(group)
  end

  # removes the group from the structure
  def remove(group)
    @groupList.delete(group)
  end
end

File: java/Composite/src/Structure.java
import java.util.ArrayList;
import java.util.List;

//Collection/group of blocks
public class Structure implements Group {
	// Collection of child group.
	private List<Group> groupList = new ArrayList<Group>();

	// assemble method should treat both single and group of blocks same.
	public void assemble() {
		
		for (Group group : groupList) {
			group.assemble();
		}
		System.out.println("Creating a structure from Group objects.");
	}

	// Adds the group to the structure.
	public void add(Group group) {
		(groupList).add(group);
	}

	// Removes the group from the structure.
	public void remove(Group group) {
		(groupList).remove(group);
	}
}